driver/src/contracts/stablex_contract.rs

```diff
@@ -189,6 +200,29 @@ impl<'a> StableXContract for StableXContractImpl<'a> {
 
         Ok(())
     }
+
+    fn stream_events(
+        &self,
+    ) -> BoxStream<'static, Result<Event<batch_exchange::Event>, ExecutionError>> {
+        Box::pin(
+            self.instance
+                .all_events()
+                .from_block(BlockNumber::Earliest)
+                .stream(),
+        )
+    }
+
+    fn past_events(
+        &self,
+    ) -> BoxFuture<'static, Result<Vec<Event<batch_exchange::Event>>, ExecutionError>> {
+        Box::pin(
+            self.instance
+                .all_events()
+                .from_block(ethcontract::BlockNumber::Earliest)
+                .to_block(ethcontract::BlockNumber::Latest)
+                .query(),
+        )
+    }
```

Comment:
In the future we will probably implement a better way to get past events so that they are not all in returned in one call. But not sure whether we do that here or in ethcontract so I am using this for now.

Reply:
Yeah, its a tough problem. For me to get it to work with Infura, I had to pick a block range and just chain a bunch of calls to get these pages one a time.

Reply:
Interesting question if this is something that could be abstracted away in ethcontracts (maybe for now we implement the logic here, but could push it upstream later?).

How do the nodes behave if we query too much data? Do they just cap at 10k event or do they error out? Is this node-dependent or rather standard? We should probably address this issue before merging this PR.

Reply:
> How do the nodes behave if we query too much data? Do they just cap at 10k event or do they error out?

Infura errors out, not sure about our parity nodes.

Reply:
> this is something that could be abstracted away in ethcontracts 

I think that is where the implementation should live. I'm going to work on adding it there so it can be used here.